content/blog/using-apache-spark-dataframes-for-processing-of-tabular-data.md

```diff
@@ -0,0 +1,362 @@
+
+## Software
+
+This tutorial will run on the MapR v5.0 Sandbox, which includes Spark 1.3
+
+* You can download the code and data to run these examples from here:
+
+  * <a target='\_blank'  href='https://github.com/caroljmcdonald/sparkdataframeexample'>https://github.com/caroljmcdonald/sparkdataframeexample</a>
+* The examples in this post can be run in the spark-shell, after launching with the spark-shell command.
+* You can also run the code as a standalone application as described in the tutorial on [Getting Started with Spark on MapR Sandbox](https://developer.hpe.com/blog/getting-started-with-spark-on-mapr-sandbox/).
+
+**The sample data sets**
+
+We will use two example datasets - one from <a target='\_blank'  href='http://www.modelingonlineauctions.com/datasets'>eBay online auctions</a> and one from the <a target='\_blank'  href='https://data.sfgov.org/Public-Safety/SFPD-Incidents-from-1-January-2003/tmnf-yvry'>SFPD Crime Incident Reporting system</a>.
+
+The eBay online auction dataset has the following data fields:
+
+***auctionid*** - unique identifier of an auction\
+***bid*** - the proxy bid placed by a bidder\
+***bidtime*** - the time (in days) that the bid was placed, from the start of the auction\
+***bidder*** - eBay username of the bidder\
+***bidderrate*** - eBay feedback rating of the bidder\
+***openbid*** - the opening bid set by the seller\
+***price*** - the closing price that the item sold for (equivalent to the second highest bid + an increment)
+
+The table below shows the data fields with some sample data:
+
+![](/img/blog_sparkdataframes_table1.png)
+
+Using Spark DataFrames we will explore the data with questions like:
+
+* How many auctions were held?
+* How many bids were made per item?
+* What's the minimum, maximum, and average number of bids per item?
+* Show the bids with price > 100
+
+The table below shows the SFPD data fields with some sample data:
+
+![](/img/blog_sparkdataframes_table2.png)
+
+Using Spark DataFrames, we will explore the SFPD data with questions like:
+
+* What are the top 10 Resolutions?
+* How many Categories are there?
+* What are the top 10 incident Categories?
+
+## Loading data into Spark DataFrames
+
+Log into the MapR Sandbox, as explained in [Getting Started with Spark on MapR Sandbox](https://developer.hpe.com/blog/getting-started-with-spark-on-mapr-sandbox/), using userid user01, password mapr. Copy the sample data files to your sandbox home directory /user/user01 using scp. Start the spark shell with:\
+`$ spark-shell`
+
+First, we will import some packages and instantiate a sqlContext, which is the entry point for working with structured data (rows and columns) in Spark and allows the creation of DataFrame objects.\
+(In the code boxes, <font color="green">comments are in Green</font> and <font color="#005CB9">output is in Blue</font>)
+
+<pre>
```

Comment:
Should all this be in a code block?
![image](https://user-images.githubusercontent.com/77496261/137518225-50f9c01c-9147-4840-a34f-1332b4db9bf3.png)

Reply:
I would have done code blocks, but the original author specifically wanted the blue/green highlighting, which we couldn't do with code blocks.